hack/lib/golang.sh

```diff
@@ -232,9 +232,18 @@ kubeedge::golang::build_binaries() {
   for bin in ${binaries[@]}; do
     echo "building $bin"
     local name="${bin##*/}"
-    set -x
-    go build -o ${KUBEEDGE_OUTPUT_BINPATH}/${name} -gcflags="${gogcflags:-}" -ldflags "${goldflags:-}" $bin
-    set +x
+    # if env KUBEEDGE_COMPILE_WITH_IMAGE is set true, we build with docker images
+    if [ ${KUBEEDGE_COMPILE_WITH_IMAGE} = true ]; then
+      set -x
+      docker run --rm -e CGO_ENABLED=0 -v ${KUBEEDGE_ROOT}:${MOUNTPATH} -w ${MOUNTPATH} ${GOLANG_IMAGE} \
+          /usr/local/go/bin/go build -o ${MOUNTPATH}/_output/local/bin/${name} -gcflags="${gogcflags:-}" -ldflags "${goldflags:-}" $bin
```

Comment:
it's better to use this image: `kubeedge/build-tools`, so all the compilation work can in one env.